nvflare/fuel/utils/fobs/fobs.py

```diff
@@ -55,22 +61,59 @@ def register(decomposer: Union[Decomposer, Type[Decomposer]]) -> None:
         decomposer: The decomposer type or instance
     """
 
-    name = _get_type_name(decomposer.supported_type())
-    if name in _decomposers:
-        return
-
     if inspect.isclass(decomposer):
         instance = decomposer()
     else:
         instance = decomposer
 
+    name = _get_type_name(instance.supported_type())
+    if name in _decomposers:
+        return
+
     if not isinstance(instance, Decomposer):
         log.error(f"Class {instance.__class__} is not a decomposer")
         return
 
     _decomposers[name] = instance
 
 
+def register_data_classes(*data_classes: Type[T]) -> None:
+    """Register generic decomposers for data classes
+
+    Args:
+        data_classes: The classes to be registered
+    """
+
+    for data_class in data_classes:
+        decomposer = DataClassDecomposer(data_class)
+        register(decomposer)
+
+
+def register_enum_types(*enum_types: Type[Enum]) -> None:
+    """Register generic decomposers for enum classes
+
+    Args:
+        enum_types: The enum classes to be registered
+    """
+
+    for enum_type in enum_types:
+        if not issubclass(enum_type, Enum):
+            raise TypeError(f"Can't register class {enum_type}, which is not a subclass of Enum")
+        decomposer = EnumTypeDecomposer(enum_type)
+        register(decomposer)
+
+
+def auto_register_enum_types(enabled=True) -> None:
```

Comment:
Why do we need to have this _enum_auto_register True or False? Instead of always True?

Reply:
This flag is just a safety measure so we can go back to the old behavior with a single switch. We may remove this flag in a later release.